helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -97,10 +99,25 @@ public ParticipantManager(HelixManager manager, HelixZkClient zkclient, int sess
   }
 
   /**
-   * Handle new session for a participang.
-   * @throws Exception
+   * Handles a new session for a participant. The new session's id is passed in when participant
+   * manager is created, as it is required in
+   * {@link org.apache.helix.manager.zk.zookeeper.ZkClient#createEphemeral(String, Object, String)}
+   * to prevent ephemeral node creation from session race condition: ephemeral node is created by an
+   * expired or unexpected session.
+   *
+   * @throws Exception if any exception occurs
    */
   public void handleNewSession() throws Exception {
```

Comment:
I might have been confused, but shouldn't we have a new handleNewSession method that takes a session Id as parameter?